MacrobenchmarkSample/macrobenchmark/src/main/java/com/example/macrobenchmark/TrivialBaselineProfileBenchmark.kt

```diff
@@ -63,14 +61,14 @@ class TrivialBaselineProfileBenchmark {
      * from the first start.
      */
     @Test
-    fun appStartupAndUserJourneys() =
-        baselineProfileRule.collectBaselineProfile(packageName = TARGET_PACKAGE, profileBlock = {
+    fun appStartupAndUserJourneys() {
+        baselineProfileRule.collectBaselineProfile(packageName = TARGET_PACKAGE) {
             startActivityAndWait()
             with(device) {
                 findObject(By.text("RECYCLERVIEW")).clickAndWait(Until.newWindow(), 500L)
                 with(
```

Comment:
I have a feeling that these nested `with` may create more confusion than simplification.

It would be very easy to have wrong behavior when writing `findObject` within the nested `with`.
It woul call `UIObject2.findObject` instead of `UiDevice.findObject`.

How about different scope functions that don't override `this` and therefore remove ambiguity (e.g. `let`)?

Reply:
`run` it is